designs/lily/src/front.mjs

```diff
@@ -0,0 +1,610 @@
+    points.crotchSeamCurveStart = points.waistIn.shiftFractionTowards(
+      points.cfSeat,
+      options.crotchSeamCurveStart
+    )
+    points.crotchSeamCurveMax = utils.beamsIntersect(
+      points.waistIn,
+      points.cfSeat,
+      points.fork,
+      points.fork.shift(0, 666)
+    )
+    points.crotchSeamCurveCp1 = points.fork
+      .shiftFractionTowards(points.crotchSeamCurveMax, options.crotchSeamCurveBend)
+      .rotate(options.crotchSeamCurveAngle * -1, points.fork)
+    points.crotchSeamCurveCp2 = points.crotchSeamCurveStart.shiftFractionTowards(
+      points.crotchSeamCurveMax,
+      options.crotchSeamCurveBend
+    )
+    points.forkCp1 = points.crotchSeamCurveCp1.rotate(90, points.fork)
+  }
+  /*
+   * Helper method to calculate the inseam delta
+   */
+  const inseamDelta = () => drawInseam().length() - store.get('inseamBack')
+  /*
+   * Helper method to calculate the outseam delta
+   */
+  const outseamDelta = () => drawOutseam().length() - store.get('outseamBack')
+  /*
+   * Helper method to lengthen/shorten both inseam and outseam
+   */
+  const adaptInseamAndOutseam = () => {
+    let shift = [
+      'kneeInCp1',
+      'kneeInCp2',
+      'kneeOutCp1',
+      'kneeOutCp2',
+      'kneeIn',
+      'kneeOut',
+      'knee',
+      'floorInCp2',
+      'floorIn',
+      'floorOutCp2',
+      'floorOut',
+      'floor',
+      'grainlineBottom',
+    ]
+    let delta = seamDelta()
+    let run = 0
+    do {
+      run++
+      for (const i of shift) points[i] = points[i].shift(90, delta)
+      delta = seamDelta()
+    } while (Math.abs(delta) > 1 && run < 10)
+  }
+  /*
+   * Helper method to determine the delta common when both inseam and outseam
+   * are either too long or too short
+   */
+  const seamDelta = () => {
+    let inseam = inseamDelta()
+    let outseam = outseamDelta()
+    return Math.abs(inseam) > Math.abs(outseam) ? outseam : inseam
+  }
+  /*
+   * Helper method that can fit either inseam or outseam
+   */
+  const adaptSeam = (side) => {
+    const out = side === 'out' ? true : false
+    let rotate = [
+      'cfSeat',
+      'crotchSeamCurveCp1',
+      'crotchSeamCurveCp2',
+      'crotchSeamCurveStart',
+      'waistIn',
+      'cfWaist',
+      'waistOut',
+    ]
+    rotate.push(out ? 'seatOut' : 'fork')
+    const deltaMethod = out ? outseamDelta : inseamDelta
+    let run = 0
+    let delta = deltaMethod()
+    do {
+      for (const i of rotate)
+        points[i] = points[i].rotate(
+          (delta / 10) * (out ? 1 : -1),
+          points[out ? 'fork' : 'seatOut']
+        )
+      run++
+      delta = deltaMethod()
+      console.log('delta', delta)
+    } while (Math.abs(delta) > 1 && run < 20)
+    console.log('run', run)
+  }
+  const adaptOutseam = () => adaptSeam('out')
+  const adaptInseam = () => adaptSeam('in')
+
+  console.log('test')
+
+  // NOTE: majority of points re-used from titan
+
+  // shape at the ankle (unlike titan)
+  let halfAnkle = store.get('halfAnkle')
```

Comment:
Use a `const` here

Reply:
Why are you storing this in a variable, when you can just read it from the store?
Looks like you are only using this once, so it's pure overhead.

Reply:
I think I was still aiming for minimal changes w.r.t. titan when I wrote that, and thought it would be easier to spot the differences that way. I've changed it now.